src/resources/views/crud/fields/relationship/fetch_or_create.blade.php

```diff
@@ -15,6 +15,14 @@
     if (!empty($current_value) || is_int($current_value)) {
         switch (gettype($current_value)) {
             case 'array':
+                // check if field have `row_key` set, if it has we are in a repeatable
+                // and should get the current row value
+                if(isset($field['row_key'])) { 
+                    if (isset($current_value[$field['row_key']]) && isset($current_value[$field['row_key']][$field['name']])) {
+                        $current_value = array($current_value[$field['row_key']][$field['name']]);
+                    }
+                }
+                
                 $current_value = $connected_entity
```

Comment:
@pxpm 
You expect here $current_value to be an array of ids.
I logged it and that's what I got.
Sometimes its just array of ids, sometimes its array of arrays.
Why? I don't know. 

```php
[2021-12-22 22:07:41] local.DEBUG: array (
  0 => '1',
)  
[2021-12-22 22:07:41] local.DEBUG: array (
  0 => '2',
)  
[2021-12-22 22:07:41] local.DEBUG: array (
  0 => 
  array (
    'characters' => '1',
    'as' => 'noooo',
  ),
  1 => 
  array (
    'characters' => '2',
    'as' => NULL,
  ),
)  

```

So obviously, `->whereIn($connected_entity_key_name, $current_value)` will throw an error. Because $current_value is not array of ids but multidimensional array. 
I made changes to relationship_select.blade.php in order to make it work.
Don't take this code too seriously.


```php
case 'array':
// check if field have `row_key` set, if it has we are in a repeatable
// and should get the current row value
if(isset($field['row_key'])) {
if (isset($current_value[$field['row_key']]) && isset($current_value[$field['row_key']][$field['name']])) {
$current_value = array($current_value[$field['row_key']][$field['name']]);
}
}

/// Check if array is multidimensional.

if (count($current_value) == count($current_value, COUNT_RECURSIVE))
{
/// Do as usual. Array contains only ids.
$current_value = $connected_entity
->whereIn($connected_entity_key_name, $current_value)
->get()
->pluck($field['attribute'], $connected_entity_key_name);
}
else
{
/// We Have to loop throught array because its array of arrays.
/// Then get $field['name'] from array and create array of ids.
$arr = array();
foreach ($current_value as $key => $value) {

array_push($arr, $value[$field['name']]);
}

$current_value = $connected_entity
->whereIn($connected_entity_key_name, $arr)
->get()
->pluck($field['attribute'], $connected_entity_key_name);
}

break;
```

Reply:
Thank you @PitchRE 🙏